### PR TITLE
Prop types to components

### PR DESCRIPTION
**Pull request by EramM805**
Added PropTypes to LikeButton and BookmarkButton
Wrote unit tests for both as well

### CODE REVIEW

File: __tests__/BookmarkButton-test.js [hunk 1]
```diff
@@ -4,54 +4,110 @@ import BookmarkButton from '@screens/app/Components/BookmarkButton/BookmarkButto
...
-  expect(tree).toMatchSnapshot();
+var assert = require('assert');
+
+test('like button snapshot test', () => {
+  const treeTrue = renderer
+    .create(<BookmarkButton isBookmarked={true} />)
+    .toJSON();
+  expect(treeTrue).toMatchSnapshot();
+  const treeFalse = renderer
+    .create(<BookmarkButton isBookmarked={false} />)
```
Comment (bujars) [inline __tests__/BookmarkButton-test.js:17]: Should we pass in values? Or allow it to just throw warnings? 

Or do we show both versions?

File: __tests__/BookmarkButton-test.js [hunk 2]
```diff
@@ -4,54 +4,110 @@ import BookmarkButton from '@screens/app/Components/BookmarkButton/BookmarkButto
...
+    'undefined value given to isBookmarked',
+  );
+  testPropTypesWithErrors(
+    null,
+    'Failed prop type: The prop `isBookmarked` is marked as required in `<<anonymous>>`, but its value is `null`.',
+    'null value given to isBookmarked',
+  );
+  testPropTypesWithErrors(
+    '',
+    'Failed prop type: Invalid prop `isBookmarked` of type `string` supplied to `<<anonymous>>`, expected `boolean`.',
```
Comment (bujars) [inline __tests__/BookmarkButton-test.js:95]: Just a side note, odd that me and you get different error messages (Possibly we should look into why that is in the future)

File: __tests__/BookmarkButton-test.js [hunk 3]
```diff
@@ -4,54 +4,110 @@ import BookmarkButton from '@screens/app/Components/BookmarkButton/BookmarkButto
...
+    ? 'bookmarked'
+    : 'notBookmarked';
+  it('testing handleSaveClick function - initially ' + bookmarkedStatus, () => {
+    const wrapper = shallow(
+      <BookmarkButton isBookmarked={initialBookmarkedStatus} />,
+    );
+    const instance = wrapper.instance();
+    const previousBookmarkedValue = instance.state.isBookmarked;
+    instance.handleSaveClick();
+    const currentBookmarkedValue = instance.state.isBookmarked;
```
[the remark this answers is not in the document]
Comment (EramM805) [reply __tests__/BookmarkButton-test.js:33]: We have a state value cuz we change states when clicking the button

File: __tests__/BookmarkButton-test.js [hunk 4]
```diff
@@ -4,54 +4,110 @@ import BookmarkButton from '@screens/app/Components/BookmarkButton/BookmarkButto
...
-  });
-  const undefinedWrapper = shallow(<BookmarkButton isBookmarked={undefined} />);
-  const undefinedIconButton = undefinedWrapper.find(IconButton).at(0);
-  it('bookmark filled when isLiked is undefined ', () => {
-    const bookmarkValue = undefinedIconButton.props().icon;
-    expect(bookmarkValue).toEqual(bookmarkOutline);
+  const colorBlack = 'black';
+  const colorGrey = 'grey';
+  it(
+    'testing icon button heart values (No errors) where isBookmarked is ' +
```
[the remark this answers is not in the document]
Comment (EramM805) [reply __tests__/BookmarkButton-test.js:49]: Yes, need to fix that. Thank you!

File: __tests__/BookmarkButton-test.js [hunk 5]
```diff
@@ -4,54 +4,110 @@ import BookmarkButton from '@screens/app/Components/BookmarkButton/BookmarkButto
...
+    let result = checkPropTypes(
+      BookmarkButton.propTypes,
+      {isBookmarked: isBookmarked},
+      'prop',
+      BookmarkButton.isBookmarked,
+    );
+    console.log(result);
+    assert(result === errorMsg);
   });
+}
```
[the remark this answers is not in the document]
Comment (EramM805) [reply __tests__/BookmarkButton-test.js:78]: I was thinking about creating a script file for it. Could be a card for the next sprint?

File: __tests__/BookmarkButton-test.js (hunk 2 above)
Comment (EramM805) [reply __tests__/BookmarkButton-test.js:95]: Yes, we should!

File: __tests__/BookmarkButton-test.js (hunk 1 above)
Comment (EramM805) [reply __tests__/BookmarkButton-test.js:17]: I'm not sure what you mean by this?

Comment (bujars) [reply __tests__/BookmarkButton-test.js:17]: So i was just asking if we maintain a snapshot test of no prop-type pushed entered, even though we require one to be. 

Like will it benefit us to have a 'broken' snapshot test?

Comment (EramM805) [reply __tests__/BookmarkButton-test.js:17]: I've added it. It throws a warning

**Pull request merged**